vendor/github.com/longhorn/backupstore/deltablock.go

```diff
@@ -996,11 +996,10 @@ func restoreBlocks(ctx context.Context, bsDriver BackupStoreDriver, deltaOps Del
 		for {
 			select {
 			case <-ctx.Done():
-				logrus.Infof("Closing goroutine for restoring blocks for volume %v", volumeName)
+				err = fmt.Errorf(types.ErrorMsgRestoreCancelled+" since server stop for volume %v", volumeName)
 				return
 			case <-deltaOps.GetStopChan():
-				logrus.Infof("Closing goroutine for restoring blocks for %v since received stop signal", volumeName)
-				err = fmt.Errorf("restoration is cancelled since received stop signal")
+				err = fmt.Errorf(types.ErrorMsgRestoreCancelled+" since received stop signal for volume %v", volumeName)
```

Comment:
If the restoration is complete, `r.stopChan` is closed and the goroutine is terminated by `case <-deltaOps.GetStopChan():`.

Reply:
IIUC, then error `err = fmt.Errorf(types.ErrorMsgRestoreCancelled+" since received stop signal for volume %v", volumeName)` will be returned even if the restoration is complete

Reply:
Ah. Let me check!

Reply:
@shuo-wu Sorry, my explanation is wrong.

The goroutine is terminated when the block channel is closed ([codes](https://github.com/longhorn/backupstore/blob/master/deltablock.go#L1004)). Block channel is closed after populating the blocks ([codes](https://github.com/longhorn/backupstore/blob/master/deltablock.go#L936-L954)).

Hence, the stop channel is only for force stop.

Reply:
Do you have any concern?

Reply:
Looks good!